app/frontend/modules/search.js

```diff
@@ -119,10 +119,15 @@ export const suggestion = (value, options) => {
 
 export const enhanceOption = (option) => {
   return {
-    name: option.label,
+    text: option.text,
+    name: getSearchableName(option),
     synonyms: (option.getAttribute('data-synonyms') ? option.getAttribute('data-synonyms').split('|') : []),
     append: option.getAttribute('data-append'),
     hint: option.getAttribute('data-hint'),
     boost: parseFloat(option.getAttribute('data-boost')) || 1
   }
 }
+
+export const getSearchableName = (option) => {
+  return option.getAttribute('data-hint') ? option.text + ' ' + option.getAttribute('data-hint') : option.text
+}
```

Comment:
how come we're making stuff out of html elements here (that's what we're doing right?)

Reply:
Kinda, we're using stimulus controller and that's how we pass the data through to the javascript layer and we're just sticking things together here. We could contract this in our ruby code, but I feel like it's not that complex